src/databricks/labs/blueprint/wheels.py

```diff
@@ -215,22 +215,61 @@ class WheelsV2(AbstractContextManager):
 
     __version: str | None = None
 
-    def __init__(self, installation: Installation, product_info: ProductInfo, *, verbose: bool = False):
+    def __init__(
+        self, installation: Installation, product_info: ProductInfo, *, verbose: bool = False, no_deps: bool = True
+    ):
         self._installation = installation
         self._product_info = product_info
         self._verbose = verbose
+        self._no_deps = no_deps
 
     def upload_to_dbfs(self) -> str:
         """Uploads the wheel to DBFS location of installation and returns the remote path."""
-        with self._local_wheel.open("rb") as f:
-            return self._installation.upload_dbfs(f"wheels/{self._local_wheel.name}", f)
+        main_wheel = []
+        for wheel in self._local_wheel:
+            if self._product_info.product_name() in wheel.name:
+                main_wheel.append(wheel)
+        if len(main_wheel) == 1:
+            with main_wheel[0].open("rb") as f:
+                return self._installation.upload_dbfs(f"wheels/{main_wheel[0].name}", f)
+        raise ValueError(
+            f"Expected one wheel containing product name {self._product_info.product_name()}, got {main_wheel}"
+        )
 
     def upload_to_wsfs(self) -> str:
         """Uploads the wheel to WSFS location of installation and returns the remote path."""
-        with self._local_wheel.open("rb") as f:
-            remote_wheel = self._installation.upload(f"wheels/{self._local_wheel.name}", f.read())
+        # TODO: This function should be modified as there is no guarantee that the main wheel.name contains product name in it - anyway the input is a list of wheels
```

Comment:
This is incorrect! This PR must not modify any other functions than _build_wheel. You only have to add a new method